net-im/telegram-desktop/files/tdesktop-3.4.8-jemalloc-only-telegram.patch

```diff
@@ -0,0 +1,37 @@
+Only link jemalloc for the Telegram binary
+
+Some combination of factors is making the different codegen tools hang when
+jemalloc is linked for those, and they're ran under portage's sandbox. Since
+this is only used during build-time, and jemalloc is merely necessary to
+improve runtime memory use, it's unnecessary to use it for anything else.
+
+--- tdesktop-3.4.8-full.orig/Telegram/CMakeLists.txt
++++ tdesktop-3.4.8-full/Telegram/CMakeLists.txt
+@@ -1331,6 +1331,12 @@ elseif (APPLE)
+         endif()
+     endif()
+ else()
++    target_link_libraries(Telegram
++    PRIVATE
++        desktop-app::external_jemalloc
++        ${CMAKE_DL_LIBS}
++    )
++
+     if (NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
+         target_link_libraries(Telegram
+         PRIVATE
+--- tdesktop-3.4.8-full.orig/cmake/options_linux.cmake
++++ tdesktop-3.4.8-full/cmake/options_linux.cmake
+@@ -63,12 +63,6 @@ if (DESKTOP_APP_SPECIAL_TARGET)
+     target_link_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto -fuse-linker-plugin>)
+ endif()
+ 
+-target_link_libraries(common_options
+-INTERFACE
+-    desktop-app::external_jemalloc
+-    ${CMAKE_DL_LIBS}
```

Comment:
Don't remove `CMAKE_DL_LIBS` here, don't add `CMAKE_DL_LIBS` to the Telegram target